tests/components/combined_energy/test_combined_energy.py

```diff
@@ -0,0 +1,104 @@
+"""Tests for setup and unload of component."""
+from __future__ import annotations
+
+from unittest.mock import ANY, AsyncMock, Mock
+
+from combined_energy.exceptions import CombinedEnergyAuthError, CombinedEnergyError
+import pytest
+
+from homeassistant.components import combined_energy
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+
+@pytest.fixture
+def config_entry():
+    """Generate a config entry for tests."""
+    return ConfigEntry(
+        1,
+        combined_energy.DOMAIN,
+        "Test",
+        data={
+            combined_energy.CONF_USERNAME: "user@example.test",
+            combined_energy.CONF_PASSWORD: "1234luggage",
+            combined_energy.CONF_INSTALLATION_ID: 999999999,
+        },
+        source="TestSource",
+        entry_id="beef",
+    )
+
+
+@pytest.fixture
+def mock_api_instance(monkeypatch):
+    """Generate a mock CombinedEnergy API."""
+    mock_api_instance = AsyncMock(combined_energy.CombinedEnergy)
+    mock_api_type = Mock(return_value=mock_api_instance)
+    monkeypatch.setattr(combined_energy, "CombinedEnergy", mock_api_type)
+    return mock_api_instance
+
+
+async def test_async_setup_entry__where_component_is_successfully_registered(
+    hass: HomeAssistant, config_entry: ConfigEntry, monkeypatch, installation
+):
+    """Check that component is successfully configured and registered."""
+    mock_api_instance = AsyncMock(
+        combined_energy.CombinedEnergy,
+        installation_id=99999,
+        installation=AsyncMock(return_value=installation),
+    )
+    mock_api_type = Mock(return_value=mock_api_instance)
+    monkeypatch.setattr(combined_energy, "CombinedEnergy", mock_api_type)
+
+    actual = await combined_energy.async_setup_entry(hass, config_entry)
+
+    assert actual is True
+    mock_api_type.assert_called_with(
+        mobile_or_email="user@example.test",
+        password="1234luggage",
+        installation_id=999999999,
+        session=ANY,
+    )
+    assert (
+        hass.data["combined_energy"]["beef"][combined_energy.DATA_API_CLIENT]
+        == mock_api_instance
+    )
+    assert (
+        hass.data["combined_energy"]["beef"][combined_energy.DATA_INSTALLATION]
+        is installation
+    )
```

Comment:
You are not supposed to access hass.data within test
https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations